community/cypher/cypher-planner-3.4/src/main/scala/org/neo4j/cypher/internal/compiler/v3_4/planner/logical/steps/AbstractIndexSeekLeafPlanner.scala

```diff
@@ -151,28 +153,33 @@ abstract class AbstractIndexSeekLeafPlanner extends LeafPlanner with LeafPlanFro
     case predicate@AsPropertySeekable(seekable: PropertySeekable)
       if seekable.args.dependencies.forall(arguments) && !arguments(seekable.ident) =>
       val queryExpression = seekable.args.asQueryExpression
-      IndexPlannableExpression(seekable.name, seekable.propertyKey, predicate, queryExpression, hints, argumentIds)
+      IndexPlannableExpression(seekable.name, seekable.propertyKey, predicate, queryExpression, hints, argumentIds, solvesPredicate = true)
 
     // ... = n.prop
     // In some rare cases, we can't rewrite these predicates cleanly,
     // and so planning needs to search for these cases explicitly
     case predicate@Equals(a, Property(seekable@LogicalVariable(_), propKeyName))
       if a.dependencies.forall(arguments) && !arguments(seekable) =>
       val expr = SingleQueryExpression(a)
-      IndexPlannableExpression(seekable.name, propKeyName, predicate, expr, hints, argumentIds)
+      IndexPlannableExpression(seekable.name, propKeyName, predicate, expr, hints, argumentIds, solvesPredicate = true)
 
     // n.prop STARTS WITH "prefix%..."
     case predicate@AsStringRangeSeekable(seekable) =>
       val partialPredicate = PartialPredicate(seekable.expr, predicate)
       val queryExpression = seekable.asQueryExpression
       val propertyKey = seekable.propertyKey
-      IndexPlannableExpression(seekable.name, propertyKey, partialPredicate, queryExpression, hints, argumentIds)
+      IndexPlannableExpression(seekable.name, propertyKey, partialPredicate, queryExpression, hints, argumentIds, solvesPredicate = true)
 
     // n.prop <|<=|>|>= value
     case predicate@AsValueRangeSeekable(seekable) =>
       val queryExpression = seekable.asQueryExpression
       val keyName = seekable.propertyKeyName
-      IndexPlannableExpression(seekable.name, keyName, predicate, queryExpression, hints, argumentIds)
+      IndexPlannableExpression(seekable.name, keyName, predicate, queryExpression, hints, argumentIds, solvesPredicate = true)
+
+    case predicate@AsDistanceSeekable(seekable) =>
+      val queryExpression = seekable.asQueryExpression
+      val keyName = seekable.propertyKeyName
+      IndexPlannableExpression(seekable.name, keyName, predicate, queryExpression, hints, argumentIds, solvesPredicate = false)
```

Comment:
I'm struggling to understand why this last case does not solve the predicate. Maybe we can add a comment to make it clearer?

Reply:
Sounds good. We could add a comment like:

> The planned index seek will _almost_ satisfy the predicate, but with the possibility of some false positives. Since it reduces the cardinality to almost the level of the predicate, we can use the predicate to calculate cardinality, but not mark it as solved, since the planner will still need to solve it with a Filter.